src/services/raspberrydevconn/app.py

```diff
@@ -0,0 +1,79 @@
+import io
+import logging
+import cherrypy
+
+from common.WIOTRestApp import *
+from common.SettingsManager import *
+from common.SettingsNode import *
+from common.RESTBase import RESTBase
+from common.CatalogRequest import *
+from common.WIOThread import WIOThread
+
+class RaspberryDevConnAPI(RESTBase):
+
+    def __init__(self, upperRESTSrvcApp, settings: SettingsNode) -> None:
+        super().__init__(upperRESTSrvcApp, 0)
+        self._catreq = CatalogRequest(self.logger, settings)
+        self._th = WIOThread(target=self._airhumidity, name="Air Humidity Thread")
+        self._th1 = WIOThread(target=self._airtemperature, name="Air Temperature Thread")
+        self._th2 = WIOThread(target=self._terrainhumidity, name="Terrain Humidity Thread")
+        upperRESTSrvcApp.subsribe_evt_stop(self._th.stop)
+        upperRESTSrvcApp.subsribe_evt_stop(self._th1.stop)
+        upperRESTSrvcApp.subsribe_evt_stop(self._th2.stop)
+        self._th.run()
+        self._th1.run()
+        self._th2.run()
+
+    def _airhumidity(self):
+        while not self._th.is_stop_requested:
+            self._th.wait(5)                                                   
+            self._catreq.publishMQTT("RaspberryDevConn", "/airhumidity", "airhumiditypayload")
+    
+    def _airtemperature(self):
+        while not self._th.is_stop_requested:
+            self._th.wait(5)                                                   
+            self._catreq.publishMQTT("RaspberryDevConn", "/airtemperature", "airtemperaturepayload")
+        
+    def _terrainhumidity(self):
+        while not self._th.is_stop_requested:
+            self._th.wait(5)                                                   
+            self._catreq.publishMQTT("RaspberryDevConn", "/terrainhumidity", "terrainhumiditypayload")
+
+    @cherrypy.tools.json_out()
+    def GET(self, *path, **args):
+        if len(path) == 0:
+            return self.asjson_info("Raspberry Device Connector Endpoint")
+        elif path[0] == "airhumidity":
+            return self.asjson("airhumidityvalue")
+        elif path[0] == "airtemperature":
+            return self.asjson("airtemperature")
+        elif path[0] == "terrainhumidity":
+            return self.asjson("terrainhumidity")
+        return self.asjson("error")
+
+class App(WIOTRestApp):
+    def __init__(self) -> None:
+
+        super().__init__(log_stdout_level=logging.INFO)
+
+        try:
+
+            self._settings = SettingsManager.json2obj(SettingsManager.relfile2abs("settings.json"), self.logger)
+            self.create(self._settings, "RaspberryDevConn", ServiceType.DEVICE, ServiceSubType.RASPBERRY)
+            self.addRESTEndpoint("/")
+            self.addRESTEndpoint("/airhumidity", [EndpointParam("state")])
+            self.addRESTEndpoint("/airtemperature", [EndpointParam("state")])
+            self.addRESTEndpoint("/terrainhumidity", [EndpointParam("state")])
```

Comment:
```suggestion
            self.addRESTEndpoint("/airhumidity", [EndpointParam("state")], endpointTypeSub=EndpointTypeSub.RESOURCE))
            self.addRESTEndpoint("/airtemperature", [EndpointParam("state")], endpointTypeSub=EndpointTypeSub.RESOURCE))
            self.addRESTEndpoint("/terrainhumidity", [EndpointParam("state")], endpointTypeSub=EndpointTypeSub.RESOURCE))
```

Reply:
Thanks for the suggestion. Can you please explain it to me? I didn't get it.

Reply:
> Thanks for the suggestion. Can you please explain it to me? I didn't get it.

In the catalog there is difference between a `GENERAL` entpoint and a `RESOURCE` endpoint, where the latter one represents for example the value of a sensor. For now it doesn't have any utility but it's better to fix this now to not lose it

Reply:
Ok perfect, thanks for the explanation. So also in the thingspeak adaptor, for example, do I need need to change them to match this notation right?

Reply:
> Ok perfect, thanks for the explanation. So also in the thingspeak adaptor, for example, do I need need to change them to match this notation right?

Nope because for me a `RESOURCE` only related to `Devices` and the ThinkSpeak adaptor is a service not a device